packages/docgen/lib/get-type-annotation.js

```diff
@@ -415,9 +409,36 @@ function getParamTypeAnnotation( tag, declarationToken ) {
 	}
 
 	try {
-		/** @type {babelTypes.TSTypeAnnotation} */
-		const typeAnnotation = paramToken.typeAnnotation.typeAnnotation;
-		return getTypeAnnotation( typeAnnotation );
+		const paramType = paramToken.typeAnnotation.typeAnnotation;
+		if ( babelTypes.isIdentifier( paramToken ) ) {
+			return getTypeAnnotation( paramType );
+		} else if ( babelTypes.isRestElement( paramToken ) ) {
+			return getTypeAnnotation( paramType );
+		} else if ( babelTypes.isArrayPattern( paramToken ) ) {
+			if ( ! tag.name.includes( '.' ) ) {
+				// unqualified name
+				return getTypeAnnotation( paramType );
+			}
```

Comment:
Possible simplification:

```suggestion
		if ( babelTypes.isIdentifier( paramToken ) ||
			 babelTypes.isRestElement( paramToken ) ||
			 ( babelTypes.isArrayPattern( paramToken ) && ! tag.name.includes( '.' ) ) )
		) {
			return getTypeAnnotation( paramType );
		} else if ( babelTypes.isArrayPattern( paramToken ) ) {
		...

```